utils.go

```diff
@@ -51,15 +76,15 @@ func JoinTraceFromHeader(
 			if err != nil {
 				return nil, err
 			}
-			contextIDMap[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
+			carrier.TracerState[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
```

Comment:
I like this file because it serves to illustrate changes to the (subtle!!) implications of changes to injection/extraction APIs... but it could easily move to `examples/` for now. Per http://opentracing.io/use-cases/#opentracing-priorities, implementors are a distant second priority to users at the moment.